### PR TITLE
AWS Bindings S3 Certification Tests

### PR DESCRIPTION
**Pull request by robertojrojas**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

## Description
As part of the AWS Bindings S3 Certification Tests, new attributes were added that require documentation updates.
Also, changed the AWS Bindings S3 component status to **Stable**

## Issue reference

reference the issue: https://github.com/dapr/components-contrib/pull/2247

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/s3.md [hunk 1]
```diff
@@ -78,6 +83,70 @@ When running the Dapr sidecar (daprd) with your application on EKS (AWS Kubernet
...
+    environment:
+      - DEBUG=1
+      - DOCKER_HOST=unix:///var/run/docker.sock
+    volumes:
+      - "<PATH>/init-aws.sh:/etc/localstack/init/ready.d/init-aws.sh"  # init hook
+      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+      - "/var/run/docker.sock:/var/run/docker.sock"
+```
+
+To able to use the S3 component, the bucket to be used needs to existing prior to use, so the example above uses a [LocalStack Initialization Hook](https://docs.localstack.cloud/references/init-hooks/) Hook to setup the bucket.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/s3.md]:
```suggestion
To use the S3 component, you need to use an existing bucket. The example above uses a [LocalStack Initialization Hook](https://docs.localstack.cloud/references/init-hooks/) to setup the bucket.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/s3.md [hunk 2]
```diff
@@ -78,6 +83,70 @@ When running the Dapr sidecar (daprd) with your application on EKS (AWS Kubernet
...
+      value: "my-secret"
+    - name: region
+      value: "us-east-1"
+```
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+To able to use the S3 component, the bucket to be used needs to existing prior to use, [Follow](https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-overview.html)
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/s3.md]:
```suggestion
To use the S3 component, you need to use an existing bucket. Follow the [AWS documentation for creating a bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-overview.html).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/s3.md [hunk 3]
```diff
@@ -70,6 +70,11 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 When running the Dapr sidecar (daprd) with your application on EKS (AWS Kubernetes), if you're using a node/pod that has already been attached to an IAM policy defining access to AWS resources, you **must not** provide AWS access-key, secret-key, and tokens in the definition of the component spec you're using.  
 {{% /alert %}}
 
+
+### S3 Bucket Creation
+{{< tabs "Minio" "Localstack" "AWS" >}}
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/s3.md]:
```suggestion
{{< tabs "Minio" "LocalStack" "AWS" >}}
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/s3.md [hunk 4]
```diff
@@ -78,6 +83,70 @@ When running the Dapr sidecar (daprd) with your application on EKS (AWS Kubernet
 3. The value for `region` is not important; you can set it to `us-east-1`.
 4. Depending on your environment, you may need to set `disableSSL` to `true` if you're connecting to Minio using a non-secure connection (using the `http://` protocol). If you are using a secure connection (`https://` protocol) but with a self-signed certificate, you may need to set `insecureSSL` to `true`.
 
+{{% /codetab %}}
+
+{{% codetab %}}
+For local development, the [localstack project](https://github.com/localstack/localstack) is used to integrate AWS S3. Follow [these instructions](https://github.com/localstack/localstack#running) to run localstack.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/s3.md]:
```suggestion
For local development, the [LocalStack project](https://github.com/localstack/localstack) is used to integrate AWS S3. Follow [these instructions](https://github.com/localstack/localstack#running) to run LocalStack.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/s3.md [hunk 5]
```diff
@@ -78,6 +83,70 @@ When running the Dapr sidecar (daprd) with your application on EKS (AWS Kubernet
...
+
+To run localstack locally from the command line using Docker, use a `docker-compose.yaml` similar to the following:
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/s3.md]:
```suggestion
To run LocalStack locally from the command line using Docker, use a `docker-compose.yaml` similar to the following:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/s3.md [hunk 6]
```diff
@@ -78,6 +83,70 @@ When running the Dapr sidecar (daprd) with your application on EKS (AWS Kubernet
...
+      - DOCKER_HOST=unix:///var/run/docker.sock
+    volumes:
+      - "<PATH>/init-aws.sh:/etc/localstack/init/ready.d/init-aws.sh"  # init hook
+      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+      - "/var/run/docker.sock:/var/run/docker.sock"
+```
+
+To able to use the S3 component, the bucket to be used needs to existing prior to use, so the example above uses a [LocalStack Initialization Hook](https://docs.localstack.cloud/references/init-hooks/) Hook to setup the bucket.
+
+In order to use localstack with your S3 binding, you need to provide the `endpoint` configuration in the component metadata. The `endpoint` is unnecessary when running against production AWS.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/s3.md]:
```suggestion
To use LocalStack with your S3 binding, you need to provide the `endpoint` configuration in the component metadata. The `endpoint` is unnecessary when running against production AWS.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick grammar review - thanks!

**Review by msfussell — APPROVED** [review]
LGTM

**Review by msfussell — APPROVED** [review]
LGTM

**Comment by msfussell** [comment]
@robertojrojas - You will need to perform a new pull on Dapr v1.11 branch for the build to succeed. Thx

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**